src/commons/utils/nimon_resource_util.py

```diff
@@ -0,0 +1,104 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+"""Module to collect resource utilisation utils."""
+
+import logging
+from src.commons.utils.corio_utils import run_local_cmd
+from src.commons.utils.corio_utils import is_package_installed_local
+from src.commons.utils.cluster_utils import RemoteHost
+from src.commons import constants as cm_cmd
+from src.commons.constants import ROOT
+from config import CLUSTER_CFG
+
+LOGGER = logging.getLogger(ROOT)
+
+
+# pylint: disable-msg=too-many-statements
+def monitor_resource_utilisation(action: str):
+    """
+    start/stop monitoring resource utilisation.
+
+    :param action: start/stop collection resource_utilisation
+    """
+    cluster_obj = None
+    host, user, passwd = None, None, None
+    server_nodes = []
+    for node in CLUSTER_CFG["nodes"]:
+        if node["node_type"] == "master":
+            if not node.get("hostname", None):
+                LOGGER.critical("failed to get master details so will not able to collect system"
+                                " stats for cluster. Nodes: '%s'", CLUSTER_CFG["nodes"])
+                continue
+            host, user, passwd = node["hostname"], node["username"], node["password"]
+            server_nodes.extend(host)
+            cluster_obj = RemoteHost(host, user, passwd)
+            resp = cluster_obj.execute_command(cm_cmd.K8S_WORKER_NODES)
+            LOGGER.debug("response is: %s", str(resp))
+            worker_node = resp[1].strip().split("\n")[1:]
+            LOGGER.info("worker nodes: %s", str(worker_node))
+            server_nodes.extend(worker_node)
+    if action == "start":
+        resp = is_package_installed_local("unzip")
+        LOGGER.debug("Local response: %s", str(resp))
+        if not resp[0]:
+            resp = run_local_cmd(cm_cmd.YUM_UNZIP)
+            LOGGER.info("Local response: %s", str(resp))
+        resp = is_package_installed_local("nimon")
+        LOGGER.debug("Local response: %s", str(resp))
+        if not resp[0]:
+            resp = run_local_cmd(cm_cmd.CMD_WGET_NIMON)
+            LOGGER.info("Local response: %s", str(resp))
+            resp = run_local_cmd(cm_cmd.UNZIP_NIMON)
+            LOGGER.info("Local response: %s", str(resp))
```

Comment:
Better to log response in `run_local_command` instead here after each call.

Reply:
Logging inside run_local_cmd (command also logging inside).